src/core_atmosphere/physics/mpas_atmphys_driver_convection.F

```diff
@@ -922,6 +978,10 @@ subroutine convection_to_MPAS(configs,diag_physics,tend_physics,its,ite)
        call mpas_pool_get_array_gpu(tend_physics,'rucuten',rucuten)
        call mpas_pool_get_array_gpu(tend_physics,'rvcuten',rvcuten)
 
+ !$acc update host(cubot, cutop, xmb_total, xmb_shallow, k22_shallow, &
+ !$acc kbcon_shallow, ktop_shallow, ktop_deep, qc_cu, qi_cu, rucuten, &
+ !$acc rvcuten)
+
        do j = jts,jte
```

Comment:
This seems to be happening on CPU and then updated to GPU. Is there a reason not to do inside parallel region.

Reply:
@Pranay-Reddy-Kommera Does this have to do with the assignments coming from the ```*_p``` pointer verisons?
If I recall correctly, these pointers were calculated in the Host memory so we use Host operations to copy the values before updating them to the GPU memory:
```
    981  !$acc update host(cubot, cutop, xmb_total, xmb_shallow, k22_shallow, &
    982  !$acc kbcon_shallow, ktop_shallow, ktop_deep, qc_cu, qi_cu, rucuten, &
    983  !$acc rvcuten)
    984 
    985        do j = jts,jte
    986        do i = its,ite
    987           cubot(i)         = cubot_p(i,j)
    988           cutop(i)         = cutop_p(i,j)
    989           xmb_total(i)     = xmb_total_p(i,j)
    990           xmb_shallow(i)   = xmb_shallow_p(i,j)
    991           k22_shallow(i)   = k22_shallow_p(i,j)
    992           kbcon_shallow(i) = kbcon_shallow_p(i,j)
    993           ktop_shallow(i)  = ktop_shallow_p(i,j)
    994           ktop_deep(i)     = ktop_deep_p(i,j)
    995 
    996           do k = kts,kte
    997              qc_cu(k,i)    = qccu_p(i,k,j)
    998              qi_cu(k,i)    = qicu_p(i,k,j)
    999              rucuten(k,i)  = rucuten_p(i,k,j)
   1000              rvcuten(k,k)  = rvcuten_p(i,k,j)
   1001           enddo
   1002        enddo
   1003        enddo
   1004 !$acc update device(cubot, cutop, xmb_total, xmb_shallow, k22_shallow, &
   1005 !$acc               kbcon_shallow, ktop_shallow, ktop_deep, qc_cu, qi_cu, &
   1006 !$acc               rucuten, rvcuten)
```

Reply:
@raghuraj19 
One thing I'd have to add here, from viewing the original 7.* version of the code:
```
    946        do j = jts,jte
    947        do i = its,ite
    948           cubot(i) = cubot_p(i,j)
    949           cutop(i) = cutop_p(i,j)
    950           nca(i)   = nca_p(i,j)
    951           do k = kts, kte
    952              w0avg(k,i)    = w0avg_p(i,k,j)
    953              rqrcuten(k,i) = rqrcuten_p(i,k,j)
    954              rqscuten(k,i) = rqscuten_p(i,k,j)
    955           enddo                          
    956        enddo
    957        enddo
```
It looks like all of the final values *only* depend on the final ```jts=jte``` iteration, since it overwrites all the previous cases for each value of ```i``` & ```k```.
Should I flag this to Michael as an issue? The ```cu_kain_fritsch``` & ```cu_tiedtke``` & ```cu_ntiedtke``` all do that here in the case-list.

Reply:
@supreethms1809 : This do loop is part of the 'cu_grell_freita' scheme which is not ported onto GPU. All the *_p variables related to 'cu_grell_freita' are assumed to be on CPU. So, the do loop is not ported to GPU.
@cponder : The jts and jte values are always equal to 1. So, having j loop does not impact the code. If I am not wrong (@raghuraj19 please correct me if I am wrong), I believe the j loop is left inside to match with the code structure of WRF.

Reply:
@supreethms1809 This should answer your question here -- the righthand-side variables (```cubot_p``` etc.) all reside on the CPU here, so we're doing the assignment on the CPU side and pushing the data across to the GPU.
This will change later, when we get the ```cu_grell_freita``` scheme onto the GPU.
Can you please "Resolve converstaion" (or should I)?